store/v2/commitment/iavl/config.go

```diff
@@ -3,13 +3,15 @@ package iavl
 // Config is the configuration for the IAVL tree.
 type Config struct {
 	CacheSize              int  `mapstructure:"cache_size"`
+	EnableSortedInsert     bool `mapstructure:"enable_sorted_insert"`
 	SkipFastStorageUpgrade bool `mapstructure:"skip_fast_storage_upgrade"`
 }
 
 // DefaultConfig returns the default configuration for the IAVL tree.
 func DefaultConfig() *Config {
 	return &Config{
 		CacheSize:              1000,
+		EnableSortedInsert:     false,
```

Comment:
why is this an extra config?

Reply:
I'm not too sure about this either.  For IAVL insert/delete order matters, it's a feature of the commitment structure and should be well known to anyone using the library.  Providing a naive `sort.Slice` with `bytes.Compare` as a library feature doesn't seem worth the added complexity, of a config option and interface method.  Clear documentaiton to handle ordering in user space space seems appropriate.  SDK defaults handle this naturally with the ordered BTree cache.

Reply:
insert/delete order matters for IAVL, but why should store/v2 have a say in what that order is?  I've tried to reason about it a lot, but it just doesn't make sense to me.  Numerical byte order by key is an arbitrary order.  As long as the order from the app is deterministic, why is any sorting at all required?

Reply:
> and the sorting insertion was broken, then there was no way to replay it.

Can you clarify what you mean by _sorting insertion was broken_ ? Neither iavl v0 nor v1 support replay from changelogs at the library level, it's required that the app orchestrate it.  Given that, the current status quo seems fine.